src/raven/client.clj

```diff
@@ -260,25 +260,46 @@
     (.configure MapperFeature/AUTO_DETECT_FIELDS false)
     (.configure MapperFeature/DEFAULT_VIEW_INCLUSION false)))
 
+(def closable? (partial instance? Closeable))
+
+(defn close-http-connection
+  "
+  When having a connection pool set, the body
+  cannot be closed as it's of a different type.
+  "
+  [response]
+  (let [{:keys [body]} response]
+    (when (closable? body)
+      (.close ^Closeable body))))
+
+(defn re-throw-response
+  "
+  Throw the response in case of its status was not 2xx
+  as we don't want Sentry errors to be unnoticed.
+  "
+  [response]
+  (let [{:keys [status]} response]
+    (when-not (-> status str first (= \2))
+      (throw (ex-info (format "Sentry HTTP error") response)))))
+
 (defn perform-http-request
   [context dsn ts payload]
   (let [json-payload             (json/write-value-as-bytes payload json-mapper)
         {:keys [key secret uri]} (parse-dsn dsn)
         sig                      (sign json-payload ts key secret)]
```

Comment:
Well, having an HTTP call completely in the background (neither `d/chain` nor `try/catch`) would lead to what we have now: we're missing Sentry alerts. I believe we must wait until the call gets finished.